### PR TITLE
Kans/nested index & partial one_ones

### PR DESCRIPTION
**Pull request by kans**
Protos don't support nested arrays particularly well, so trying the stringly way!

See [example/models/city/v1/city.pb.pgdb.go](https://github.com/ductone/protoc-gen-pgdb/pull/45/files#diff-0d295ea9554e411e396f1717bb914c9fec8233f4af57c21b3af21ef60a682dcd) for Operators which look correct to me.

```
message Attractions {
  option (pgdb.v1.msg).indexes = {
    name: "furrrs"
    method: INDEX_METHOD_BTREE_GIN,
    columns: [
      "tenant_id",
      "zoo_shop.fur"
    ]
  };
  oneof what {
    models.animals.v1.Pet pet = 10;
    models.zoo.v1.Shop zoo_shop = 11;
  }
};
```

```
CREATE INDEX CONCURRENTLY IF NOT EXISTS
  "pbidx_furrrs_attractions_models_city_v1_97e8b7fb"
ON
  "pb_attractions_models_city_v1_e136cbfc"
USING
  GIN
(
  "pb$tenant_id", 
  "pb$11$fur"
)

...
=== RUN   TestSchemaAttractions
 CREATE TABLE IF NOT EXISTS
  "pb_attractions_models_city_v1_e136cbfc"
(
 "pb$11$fur" int4 DEFAULT 0,
```

### CODE REVIEW

File: proto/pgdb/v1/pgdb.proto [hunk 1]
```diff
@@ -24,7 +24,7 @@ message MessageOptions {
       INDEX_METHOD_BTREE_GIN = 3;
     }
     string name = 1;
-    IndexMethod nethod = 2;
+    IndexMethod method = 2;
```
Comment (kans) [inline proto/pgdb/v1/pgdb.proto:27]: Will this break something down stream?

Comment (pquerna) [reply proto/pgdb/v1/pgdb.proto:27]: no, we can change this.

File: internal/pgdb/pgdb_index.go [hunk 2]
```diff
@@ -55,25 +59,42 @@ func (module *Module) extraIndexes(ctx pgsgo.Context, m pgs.Message, ix *importT
...
+		resolution := ""
+		for i, p := range path {
+			f := fieldByName(message, p)
+			if i != len(path)-1 {
+				resolution += getNestedName(f)
+				message = f.Type().Embed()
+				continue
+			}
+
+			if p == tenantIdField {
```
Comment (kans) [inline internal/pgdb/pgdb_index.go:81]: This is the wrong tenantIdField.  We can not actually get the correct one here given the limited API.

File: example/models/city/v1/city.proto [hunk 3]
```diff
@@ -20,6 +20,14 @@ message Attractions {
       "numid"
     ]
   };
+  option (pgdb.v1.msg).indexes = {
+    name: "furrrs"
+    method: INDEX_METHOD_BTREE_GIN,
+    columns: [
+      "tenant_id",
+      "zoo_shop🌮fur"
```
Comment (pquerna) [inline example/models/city/v1/city.proto:28]: tacos are fun, but any reason to not use a period `.`?

Comment (kans) [reply example/models/city/v1/city.proto:28]: I like them for debugging because they are self highlighting.  I'll switch it over.

**Pull request merged**